Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -1227,6 +1234,10 @@ void RenderElement::clearChildNeedsLayout()
 void RenderElement::setNeedsSimplifiedNormalFlowLayout()
 {
     ASSERT(!isSetNeedsLayoutForbidden());
+    if (CheckedPtr box = dynamicDowncast<RenderBox>(this); box && !box->canUseOverlayScrollbars() && box->scrollsOverflow()) {
+        setNeedsLayout();
+        return;
+    }
```

Comment:
I think it is somewhat unexpected that the caller determined the renderer only needs simplified layout and yet we set regular layout here. Can't the caller figure this out and not set up the render for simplified layout in this case?

Reply:
The 'caller' in this case is quite a few different spots, but basically all coming from the style system's change hint. I don't think the style code make this determination, and this seemed like the best single-location spot to intercept it.

I could probably add a wrapper for 'set whatever layout is needed for this change hint'.

Reply:
ok, let me look.

Reply:
so this change prevents us from running simplified layout (which is simply about collecting overflow) on a renderer that is a scroll container with space taking scrollbars? I am surprised we never caught that as it's essentially `<div style="overflow: scroll"><div out-of-flow></div></div>` and script: move out-of-flow box.
Am I missing something here?

Reply:
I think it's exactly that.

The missing bit is the original/core change of this PR, that the 'out-of-flow' state flags weren't set at the point we potentially requested simplified overflow.

So that testcase wouldn't do simplified layout, despite looking like it should.